addons/hr_contract/models/hr_employee.py

```diff
@@ -332,3 +333,53 @@ def action_open_contract(self):
 
         action['res_id'] = self.contract_ids[0].id
         return action
+
+    def _get_employee_working_periods(self, employee_ids, start, stop):
+        """
+            A dedicated method to get working period values neccessary.Returns the rows by
+            adding contract details for each row of employees_rows.
+            :param employee_ids: A list of employee ids each row
+            :return: employees_rows + contract details
+        """
+        employees_sudo = self.browse(employee_ids).sudo()
+        employees_rows = {employee: {"working_periods": []} for employee in employee_ids}
+        # Contracts of employees in current scale of time
+        contracts = employees_sudo._get_contracts(start, stop, ["draft", "open", "close"])
+        # Contracts of employees who don't have active contract in current scale of time
+        employees_with_contract_outside_current_scale = dict(
+            self.env["hr.contract"].sudo()._read_group(
+                domain=[
+                    ("employee_id", "in", set(employees_sudo.ids) - set(contracts.mapped("employee_id").ids)),
```

Comment:
`mapped` is implicit for relational fields.
```suggestion
                    ("employee_id", "in", set(employees_sudo.ids) - set(contracts.employee_id.ids)),
```